internal/vendors/accessor.go

```diff
@@ -143,6 +144,74 @@ func (p *postgresVendorAccessor) GetByLocation(ctx context.Context, location str
 	return vendors, nil
 }
 
+func (p *postgresVendorAccessor) GetByProduct(ctx context.Context, product string) ([]Vendor, error) {
```

Comment:
```suggestion
func (p *postgresVendorAccessor) GetByProductParts(ctx context.Context, productParts []string) ([]Vendor, error) {
```
IMO, it would be better for the accessor to just focus on executing the queries, and let the client handle the split

Also it adds context upon what the interface wants, i.e. parts of queries, instead of a string which is less intuitive

Reply:
another alternative would just be to rename the function 😂 just to add verbosity

```suggestion
func (p *postgresVendorAccessor) GetByProductDescription(ctx context.Context, productDescription string) ([]Vendor, error) {
```

Reply:
Done, please rereview!